### ISSUE TITLE
Error "Destination path is not an empty directory" when specifying a configuration file

### ISSUE THREAD

**Issue by ppsmilesi** [lerebear/sizeup-action#55]
Hello !

Thank you for the github action and the sizeup library, it looks awesome !

I have tried to run the action with a configuration file without success.
I have checked out the config file at first step of the workflow for it to be available but then, the sizeup action has run into this error:
`Error: fatal: destination path '.' already exists and is not an empty directory.`

Output of the workflow:
```
Run lerebear/sizeup-action@v0.8.3
  with:
    token: ***
    configuration-file-path: .github/workflows/sizeup/config.yml
    git-diff-options: --ignore-space-change
Reading sizeup configuration from ".github/workflows/sizeup/config.yml"
Evaluating pull request ...
Error: fatal: destination path '.' already exists and is not an empty directory.
```

Workflow fiile `.github/workflows/sizeup.yml`:
```
name: SizeUp

on:
  pull_request:
    types:
      - opened
      - synchronize
      - reopened
      - edited
      - ready_for_review

permissions:
  contents: read
  issues: write
  pull-requests: write

jobs:
  sizeup:
    name: Estimate reviewability
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v4 # Check out the repository to be able to read config file
        with:
          sparse-checkout: |
            .github/workflows/sizeup/config.yml
          sparse-checkout-cone-mode: false

      - name: Run sizeup
        # https://github.com/lerebear/sizeup-action/blob/main/README.md#versioning
        uses: lerebear/sizeup-action@v0.8.3
        id: sizeup-action

        with:
          token: "${{ secrets.GITHUB_TOKEN }}"

          configuration-file-path: .github/workflows/sizeup/config.yml
```

Thank you for your help !

**Comment by lerebear** [comment]
Thank you for the report @ppsmilesi. I suspect that the error you've encountered is due to incompatibility between `actions/checkout` and the internal operations of this action. There is a small note on this in the README:

> Please note that the workflow configuration above does not use [actions/checkout](https://github.com/actions/checkout). This is because actions/checkout does not provide enough options to make a customized git diff command maximally efficient. Instead, this Action will perform its own clone, fetch, and checkout operations using the provided token.

Could you please try removing the first step of your workflow (the one that uses `actions/checkout`)? Once you allow this action to manage the checkout flow then I believe this should work properly.

**Comment by ppsmilesi** [comment]
Hey ! Thank you for coming back to me !

I have added the checkout step because without it, the sizeup action cannot even find the config file.

Here is the error:
```
Run lerebear/sizeup-action@v0.8.3 

with: 

token: *** 

configuration-file-path: .github/workflows/sizeup/config.yml 

git-diff-options: --ignore-space-change 

Reading sizeup configuration from ".github/workflows/sizeup/config.yml" 

Error: ENOENT: no such file or directory, open '.github/workflows/sizeup/config.yml'
```

I am pretty sure the path to the config file is correct.
Does the action clone the repo before reading the config file for it to be available ?

Thank you

**Comment by lerebear** [comment]
@ppsmilesi thank you for helping to uncover and debug this issue!

> Does the action clone the repo before reading the config file for it to be available ?

It does ordinarily, but I had unwittingly broken this behaviour in #54, which affected [0.8.3](https://github.com/lerebear/sizeup-action/releases/tag/v0.8.3). This should now be fixed in [0.8.4](https://github.com/lerebear/sizeup-action/releases/tag/v0.8.4). If you run into any further issues please let me know.

**Comment by ppsmilesi** [comment]
Thank you @lerebear ! 8.0.4 is fixed and working 🙌
